src/sys/socket/mod.rs

```diff
@@ -644,6 +644,20 @@ pub enum ControlMessage<'a> {
     /// following one by one, and the last, possibly smaller one.
     #[cfg(target_os = "linux")]
     UdpGsoSegments(&'a u16),
+
+    /// Configure the sending addressing and interface for v4
+    ///
+    /// For further information, please refer to the
+    /// [`ip(7)`](http://man7.org/linux/man-pages/man7/ip.7.html) man page.
+    #[cfg(target_os = "linux")]
+    Ipv4PacketInfo(&'a libc::in_pktinfo),
+
+    /// Configure the sending addressing and interface for v6
+    ///
+    /// For further information, please refer to the
+    /// [`ipv6(7)`](http://man7.org/linux/man-pages/man7/ipv6.7.html) man page.
+    #[cfg(target_os = "linux")]
+    Ipv6PacketInfo(&'a libc::in6_pktinfo),
```

Comment:
This feature appears to be supported by NetBSD, OSX, and FreeBSD.

Reply:
Done.